feat: ADR-54 oracle for MANA prices (#37)
decentraland/adr: docs/ADR-54-oracle-for-mana-pricing.md

# Use an Oracle for MANA pricing according to USD rate

## Statement of the problem

> Some features require the user to pay a certain amount of MANA for execution. These features include, publishing a collection, which charges the user a fixed amount of MANA depending on the rarities of the items being published with the collection, and claiming a name, were the user is charged a fixed amount of MANA to claim. There is also the purchasing of third party item slots, also priced in MANA.

Reviewer: What do you think about enumerating the cases?
```suggestion
Some features require the user to pay a certain amount of MANA for execution. These features include:
- Publishing a collection, which charges the user a fixed amount of MANA depending on the rarities of the items being published with the collection
- Claiming a name, were the user is charged a fixed amount of MANA to claim.
- Purchasing Third Party item slots, which is charged according to the amount of slots bought.
```

Given the price fluctuations of the token, more importantly the increase in price over time, these features have become more expensive for the user to consume, thus discouraging content creators from actually adding content to the platform, hurting Decentraland as a whole on the process.

> In the casing of collection publishing, prices are being changed manually each weak so they reflect a certain price in USD, making the price more stable and fair for all creators. However, this being a manual process can become tedius and not completely precise as MANA price might still fluctuate between publication price changes.

Reviewer: ```suggestion
In the case of publishing a collection, prices are being changed manually each week so they reflect a certain price in USD, making the price more stable and fair for all creators. However, this being a manual process can become tedious and not completely precise as MANA price might still fluctuate between publication price changes.
```

> ## Solution proposed

Reviewer: ```suggestion
## Proposed solution
```

Chainlink MANA/USD Data Feed allows our contracts to obtain the rate of MANA in USD which will be used to calculate the amount of MANA required to perform certain transactions given a previously defined amount of USD.

For more information about Chainlink go check their [homepage](https://chain.link/) or their [whitepaper](https://chain.link/whitepaper).

In summary, Chainlink is a solution in which via Oracles, one can obtain off-chain data in a tamper-proof and still decentralized manner to be used on-chain.

> In order to implement these price feeds we will create a new contract `ChainlinkOracle` with a `getRate` function that will the be used by other contracts which require obtaining the rate of MANA/USD to reflect a price accordingly. This `ChainlinkOracle` will receive both a Data Feed (Represented as an [AggregatorV3Interface](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/interfaces/AggregatorV3Interface.sol) by the Chainlink [docs](https://docs.chain.link/docs/get-the-latest-price/#solidity)) and the amount of decimals I want the rate to be returned with.

Reviewer: ```suggestion
In order to implement these price feeds we will create a new contract `ChainlinkOracle` with a `getRate` function that will be used by other contracts which require obtaining the rate of MANA/USD to reflect a price accordingly. This `ChainlinkOracle` will receive both a Data Feed (Represented as an [AggregatorV3Interface](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/interfaces/AggregatorV3Interface.sol) by the Chainlink [docs](https://docs.chain.link/docs/get-the-latest-price/#solidity)) and the amount of decimals the rate will be returned with.
```

# Use an Oracle for MANA pricing according to USD rate

## Statement of the problem

Some features require the user to pay a certain amount of MANA for execution. These features include:

- Publishing a collection, which charges the user a fixed amount of MANA depending on the rarities of the items being published with the collection

- Claiming a name, where the user is charged a fixed amount of MANA to claim.

- Purchasing Third Party item slots, which are charged according to the number of slots bought.

> Given the price fluctuations of the token, more importantly, the increase in price over time, these features have become more expensive for the user to consume, thus discouraging content creators from actually adding content to the platform, hurting Decentraland as a whole on the process.

Reviewer 2: I will remove this paragraph and add  just something like

```The community has been used the DAO to define the price in MANA but pegged to the USD dollar```

Author: Done

> In the case of publishing a collection, prices are being changed manually each week so they reflect a certain price in USD, making the price more stable and fair for all creators. However, this being a manual process can become tedious and not completely precise as MANA price might still fluctuate mid-week.

Reviewer 2: This is not 100% true. It has been changed manually from 500 MANA to 100 MANA once. Can you rephrase it?

Author: Done

## Proposed solution

> Chainlink MANA/USD Data Feed allows our contracts to obtain the rate of MANA in USD. This can be used, given an original price in USD, to calculate how much MANA is required by the user to perform a certain transaction.

Reviewer 2: I will add something like `in Ethereum and Polygon chains`

Author: Dome
